cmd/internal.go

```diff
@@ -246,31 +264,36 @@ func ParseLineage(pipe *pipeline.Pipeline, asset *pipeline.Asset) error {
 		if upstreamAsset == nil {
 			return fmt.Errorf("upstream asset not found: %s", upstream.Value)
 		}
-		columnMetadata[upstreamAsset.Name] = makeColumnMap(upstreamAsset.Columns)
+		if len(upstreamAsset.Columns) > 0 {
+			columnMetadata[upstreamAsset.Name] = makeColumnMap(upstreamAsset.Columns)
+		}
 	}
 
 	lineage, err := parser.ColumnLineage(asset.ExecutableFile.Content, "", columnMetadata)
 	if err != nil {
 		return fmt.Errorf("failed to parse column lineage: %w", err)
 	}
-
 	for _, lineageCol := range lineage.Columns {
 		for _, upstream := range lineageCol.Upstream {
 			upstreamAsset := pipe.GetAssetByName(upstream.Table)
 
 			if upstreamAsset == nil {
 				continue
 			}
-			upstreamCol := upstreamAsset.GetColumnWithName(upstream.Column)
-			if upstreamCol == nil {
-				continue
-			}
-
-			newCol := *upstreamCol
-			newCol.Name = lineageCol.Name
 
-			if col := asset.GetColumnWithName(lineageCol.Name); col == nil {
-				asset.Columns = append(asset.Columns, newCol)
+			if upstream.Column != "*" {
+				upstreamCol := upstreamAsset.GetColumnWithName(upstream.Column)
+				if upstreamCol == nil {
+					continue
+				}
+
+				if lineageCol.Name != "*" {
+					newCol := *upstreamCol
+					newCol.Name = lineageCol.Name
+					if col := asset.GetColumnWithName(lineageCol.Name); col == nil {
+						asset.Columns = append(asset.Columns, newCol)
+					}
+				}
```

Comment:
I'd invert these if statements:
```suggestion
			if upstream.Column == "*" {
				continue
			}
			
			if lineageCol.Name == "*" {
				continue
			}
				
			upstreamCol := upstreamAsset.GetColumnWithName(upstream.Column)
			if upstreamCol == nil {
				continue
			}
				
			newCol := *upstreamCol
			newCol.Name = lineageCol.Name
			if col := asset.GetColumnWithName(lineageCol.Name); col == nil {
				asset.Columns = append(asset.Columns, newCol)
			}
			
```

Reply:
apart from quality checks is there anything we should ignore

Reply:
I don't think so.